tensorflow_addons/image/color_ops.py

```diff
@@ -27,38 +27,47 @@
 from functools import partial
 
 
-def _scale_channel(image: TensorLike, channel: int) -> tf.Tensor:
+def _scale_channel(
+    image: TensorLike, channel: int, bins: Optional[int] = 256
+) -> tf.Tensor:
     """Scale the data in the channel to implement equalize."""
     image_dtype = image.dtype
     image = tf.cast(image[:, :, channel], tf.int32)
 
     # Compute the histogram of the image channel.
-    histo = tf.histogram_fixed_width(image, [0, 255], nbins=256)
+    histo = tf.histogram_fixed_width(image, [0, bins - 1], nbins=bins)
 
     # For the purposes of computing the step, filter out the nonzeros.
     nonzero_histo = tf.boolean_mask(histo, histo != 0)
-    step = (tf.reduce_sum(nonzero_histo) - nonzero_histo[-1]) // 255
+    if (bins - 1) == 0:
+        step = tf.reduce_sum(nonzero_histo) - nonzero_histo[-1]
+    else:
+        step = (tf.reduce_sum(nonzero_histo) - nonzero_histo[-1]) // (bins - 1)
 
     # If step is zero, return the original image.  Otherwise, build
     # lut from the full histogram and step and then index from it.
     if step == 0:
         result = image
     else:
         lut_values = (tf.cumsum(histo, exclusive=True) + (step // 2)) // step
-        lut_values = tf.clip_by_value(lut_values, 0, 255)
+        lut_values = tf.clip_by_value(lut_values, 0, bins - 1)
```

Comment:
For line 43 could it clip with 0 ?

Reply:
If I understand your question correctly, I suppose so, then all you would be left with is a Tensor of zeros. But that would also mean that the equalization in question would be in one histogram bin, which would not be the intended use case.

Reply:
Yes, probably you could test also 0, 256, and not 256 bins with https://scikit-image.org/docs/dev/api/skimage.exposure.html#skimage.exposure.equalize_hist